examples/SMAC4HPO_hyperband_mlp_instances.py

```diff
@@ -0,0 +1,156 @@
+cs.add_hyperparameters([n_layer, n_neurons, activation, solver, batch_size, learning_rate, learning_rate_init])
+
+# Adding conditions to restrict the hyperparameter space
+# Since learning rate is used when solver is 'sgd'
+use_lr = CS.conditions.EqualsCondition(child=learning_rate, parent=solver, value='sgd')
+# Since learning rate initialization will only be accounted for when using 'sgd' or 'adam'
+use_lr_init = CS.conditions.InCondition(child=learning_rate_init, parent=solver, values=['sgd', 'adam'])
+# Since batch size will not be considered when optimizer is 'lbfgs'
+use_batch_size = CS.conditions.InCondition(child=batch_size, parent=solver, values=['sgd', 'adam'])
+# We can also add  multiple conditions on hyperparameters at once:
+cs.add_conditions([use_lr, use_batch_size, use_lr_init])
+
+# Defining instances (number of folds for CV)
+instances = [['2'], ['3'], ['5'], ['10']]
+
+# SMAC scenario object
+scenario = Scenario({"run_obj": "quality",      # we optimize quality (alternative runtime)
+                     "wallclock-limit": 100,    # max duration to run the optimization (in seconds)
+                     "cs": cs,                  # configuration space
+                     "deterministic": "true",
+                     "limit_resources": False,  # Disables pynisher to pass cutoff directly to the target algorithm.
+                                                # Timeouts have to be taken care within the TA
+                     "cutoff": 10,              # Cutoff denotes the number of epochs for MLP.
+                                                # It constant across all instances
+                     "instances": instances     # Optimize across all given instances
+                     })
+
+# intensifier parameters
+# if no argument provided for budgets, hyperband decides them based on the number of instances available
+intensifier_kwargs = {'initial_budget': 1, 'max_budget': 4, 'eta': 2,
+                      'instance_order': None}  # You can also shuffle the order of using instances by this parameter
+
+# To optimize, we pass the function to the SMAC-object
+smac = SMAC4HPO(scenario=scenario, rng=np.random.RandomState(42),
+                tae_runner=mlp_from_cfg,
+                intensifier=Hyperband,                  # you can also change the intensifier to use like this!
+                                                        # This example currently uses Hyperband intensification,
+                intensifier_kwargs=intensifier_kwargs)  # all parameters related to intensifier can be passed like this
+
+# Example call of the function with default values
+# It returns: Status, Cost, Runtime, Additional Infos
+def_costs = []
+for i in instances:
+    cost = smac.get_tae_runner().run(cs.get_default_configuration(), i[0], 10, 0)[1]
+    def_costs.append(cost)
+print("Value for default configuration: %.4f" % (np.mean(def_costs)))
```

Comment:
I don' t think computing the mean across different budgets is what we want, but instead seeing how the value changes with different budgets?